### DISCUSSION TITLE
Component naming (nuxt 3)

### DISCUSSION THREAD

**Discussion by EmilyFlocc** [nuxt/nuxt#16937] [General]
In Nuxt 3, components are pre-pended with their directory name. Can you have multiple components with the same name that sit within different directories? Or is this bad practise?

For example, I have the following folder structure:

- desktop
  - Main.vue
- mobile
  - Main.vue

Which I can then call as `<DesktopMain />` and `<MobileMain />`.

This works, but I'm wondering if it's bad practise or could be causing my hydration errors? Or it could be something completely different haha, just trying to work out where I'm going wrong!

**Comment by danielroe** [comment]
Yep - this is absolutely fine!

**Comment by EmilyFlocc** [comment]
Ah great, thank you @danielroe! Just wanted to be sure I wasn't going to regret it further down the line!
